serverless-plugin/handler.template.js

```diff
@@ -1,55 +1,66 @@
 'use strict';
 
 const { spawn } = require('child_process');
+const net = require('net');
 
 function wrapper(options) {
     const executable = options['executable'];
     const execArguments = options['arguments'];
-    return function (event, context, callback) {
+    return async function (event, context) {
         process.env['PATH'] = process.env['PATH'] + ':' +
             process.env['LAMBDA_TASK_ROOT'];
 
+        // Keep track of the output result
+        let output = '';
+        const server = net.createServer(function(socket) {
+            socket.on('data', chunk => output += chunk);
+        });
+
+        const address = await new Promise(
+            resolve =>
+                server.listen({port: 0, host: "127.0.0.1"}, () => resolve(server.address())));
+        const port = address.port;
+
+        process.env['SERVERLESS_HASKELL_COMMUNICATION_PORT'] = `${port}`;
+
         const main = spawn('./' + executable, execArguments, {
-            stdio: ['pipe', process.stdout, process.stderr, 'pipe'],
+            stdio: ['pipe', process.stdout, process.stderr],
         });
         const stdin = main.stdin;
-        const communication = main.stdio[3];
 
         // Send the event to the process
         stdin.end(JSON.stringify(event) + '\n', 'utf8');
 
-        // Keep track of the process output
-        let output = '';
-        communication.on('data', chunk => output += chunk);
-
         let exited = false;
 
-        main.on('exit', function (code) {
-            if (!exited) {
-                exited = true;
-                if (code == 0) {
-                    try {
-                        const result = JSON.parse(output);
-                        callback(null, result);
-                    } catch (err) {
-                        console.error('child process output bad JSON: ' + output);
-                        callback('child process output bad JSON: ' + output);
+        const result = new Promise((resolve, reject) => {
```

Comment:
I found it a bit confusing that there are two variables called `result` (here and line 43). I know they are in different scopes, but it would be more readable if you renamed one of them.